packages/@uppy/provider-views/src/ProviderView/ProviderView.jsx

```diff
@@ -301,6 +301,9 @@ export default class ProviderView extends View {
               folder: name,
             })
           } else {
+            // TODO we don't really know at this point whether any files were actually added
+            // (only later after addFiles has been called) so we should probably rewrite this.
+            // Example: If all files fail to add due to restriction error, it will still say "Added 100 files from folder"
```

Comment:
the problem is that when you add folders, we cannot apply restrictions in the provider view, because we don't know what is inside the folders at this point (folder structure can be infinitely deep). only after `addFiles` has been called and finished, only then do we know how many files actually passed validation, so I think this TODO still stands.

Reply:
That's a good point!

Reply:
restrictions are not validated when starting the upload AFAIK. They are only validated when adding the files. I still think it's not fine to tell the user "150 files were added", while at the same time saying "failed to add folder x and y", and in reality no files got added. You can test it, and you will see how it looks very confusing. (try to select a folder containing 150 files, and then add a restriction that prevents the folder from being added, and you'll see that no files will get added)

Reply:
note that this behaviour is not new in this PR. it existed before this PR also, I just added a todo to fix it, because it's not a super easy thing to fix, so I didn't want to include a fix in this PR

Reply:
Got it, thanks. Can we create a new issue for it? I think this is more important than a todo comment. We can resolve this comment after that.

Reply:
https://github.com/transloadit/uppy/issues/4414